modules/systemdstates/collect.go

```diff
@@ -74,16 +78,33 @@ func (s SystemdStates) filterUnits(units []dbus.UnitStatus) []dbus.UnitStatus {
 }
 
 func extractUnitType(unit string) (string, error) {
-	validTypes := []string{"service", "socket", "device", "mount", "automount", "swap", "target", "path", "timer", "scope"}
 	ut := ""
-	for _, t := range validTypes {
-		if strings.HasSuffix(unit, fmt.Sprintf(".%s", t)) {
-			ut = t
-			break
-		}
+	idx := strings.LastIndexByte(unit, '.')
+
+	switch suffix := unit[idx:]; suffix {
```

Comment:
i suggest

```go
idx := strings.LastIndexByte(unit, '.')
if idx <= 0 {
	...
}
unitType := unit[idx+1:]
```

^^ that is extract already and we can return w/o switch.

I see you want to add some type validation which is fine by me.

I suggest to make it a separate function and use in `extractUnitType` 

```go
idx := strings.LastIndexByte(unit, '.')
if idx <= 0 {
	...
}
unitType := unit[idx+1:]
if !isUnitTypeValid(unitType) {
   ...
}
...
```